renku/core/dataset/dataset.py

```diff
@@ -791,11 +793,38 @@ def show_dataset(name: str, tag: Optional[str] = None):
     return DatasetDetailsJson().dump(dataset)
 
 
+def add_datadir_files_to_dataset(client: "LocalClient", dataset: Dataset) -> None:
+    """Add all files in a datasets data directory to the dataset.
+
+    Args:
+        client(LocalClient): The ``LocalClient``.
+        dataset(Dataset): The dataset to add data dir files to.
+    """
+    datadir = get_safe_relative_path(dataset.get_datadir(), client.path)
+
+    if datadir.exists():
+        # NOTE: Add existing files to dataset
+        dataset_files: List[DatasetFile] = []
+        files: List[Path] = []
+        for file in datadir.rglob("*"):
```

Comment:
This returns directories as well. You should use `renku.core.util.os.get_files`.